R/trace.R

```diff
@@ -857,7 +857,7 @@ trace_call_text <- function(call, collapsed, namespace, scope) {
   if (is_string(scope, "global")) {
     text <- paste0("global ", text)
   } else if (is_string(scope, "local") && !is_na(namespace)) {
-    text <- paste0(namespace, " ", text)
+    text <- paste0(namespace, " (local) ", text)
```

Comment:
Does it still make sense to include the namespace here?

Reply:
I think so because these functions do belong to the namespace, they are just not accessible easily.

Reply:
Do they? I'd say they belong to a function that belongs to the namespace.

Reply:
I agree but I see belonging as a transitive property. At a high level the goal of showing the package names is to make it clearer which packages are responsible for which calls. I would prefer not to hide this information for local functions.

Reply:
It just feels to me like the namespace and the (local) are fighting each other. Would you mind seeing what it looks like if you just drop the namespace for local functions and don't add (local)?

Reply:
I see what you mean. Here is what it looks like without namespace: https://github.com/r-lib/rlang/commit/62f8a16627ff6c34349a9c3fc1374cd959f3e5cc

Here is an example where I think this is a regression:

```r
f <- function() {
  helper <- function() abort("foo")
  tryCatch(helper())
}
environment(f) <- ns_env("rlang")
env_unlock(ns_env("rlang"))
env_bind(ns_env("rlang"), f = f)

f()
#> Error in `helper()`:
#> ! foo

last_error()
#> <error/rlang_error>
#> Error in `helper()`:
#> ! foo
#> ---
#> Backtrace:
#>  1. rlang:::f()
#>  4. helper()
#> Run `rlang::last_trace()` to see the full context.

last_trace()
#> rlang::last_trace()
#> <error/rlang_error>
#> Error in `helper()`:
#> ! foo
#> ---
#> Backtrace:
#>     ▆
#>  1. └─rlang:::f()
#>  2.   ├─base::tryCatch(helper())
#>  3.   │ └─tryCatchList(expr, classes, parentenv, handlers)
#>  4.   └─helper()
#>  5.     └─rlang::abort("foo")
```

Compare to the current output:

```r
last_error()
#> <error/rlang_error>
#> Error in `helper()`:
#> ! foo
#> ---
#> Backtrace:
#>  1. rlang:::f()
#>  4. rlang helper()
#> Run `rlang::last_trace()` to see the full context.

last_trace()
#> <error/rlang_error>
#> Error in `helper()`:
#> ! foo
#> ---
#> Backtrace:
#>     ▆
#>  1. └─rlang:::f()
#>  2.   ├─base::tryCatch(helper())
#>  3.   │ └─base tryCatchList(expr, classes, parentenv, handlers)
#>  4.   └─rlang helper()
#>  5.     └─rlang::abort("foo")
```

So I suggest we just keep the current behaviour, what do you think?